solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,530 @@
+                // total amount to convert to the governance token
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(amount);
+
+                networkTokenConversionAmounts[i] = amount;
+            } else {
+                if (token != NATIVE_TOKEN_ADDRESS) {
+                    // if the source token is a regular token, approve the converter to withdraw the token amount
+                    ensureAllowance(token, network, amount);
+                } else {
+                    // if the source token is actually an ETH reserve, make sure to pass its value to the network
+                    value = amount;
+                }
+
+                // perform the actual conversion and optionally send ETH to the network
+                uint256 networkTokenConversionAmount =
+                    network.convertByPath2{ value: value }(path, amount, 1, address(this));
+
+                // update network conversion amounts
+                networkTokenConversionAmounts[i] = networkTokenConversionAmount;
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(networkTokenConversionAmount);
+            }
+        }
+
+        // calculate the burn incentive fee and reduce it from the total amount to convert
+        NetNetworkTokenConversionAmounts memory netNetworkTokenConversionAmounts =
+            netNetworkConversionAmounts(grossNetworkTokenConversionAmount);
+
+        // approve the governance token converter to withdraw the network token amount
+        ensureAllowance(_networkToken, network, netNetworkTokenConversionAmounts.amount);
+
+        // convert all network token amounts to the governance token
+        totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(
+            network.convertByPath2(strategy.govPath, netNetworkTokenConversionAmounts.amount, 1, address(this))
+        );
+
+        // update the stats of the vortex
+        _totalBurnedAmount = _totalBurnedAmount.add(totalGovTokenAmountToBurn);
+
+        // burn all the converter governance tokens
+        _govTokenGovernance.burn(totalGovTokenAmountToBurn);
+
+        // transfer the incentive fee to the caller
+        _networkToken.transfer(msg.sender, netNetworkTokenConversionAmounts.incentiveFeeAmount);
+
+        emit Burned(tokens, strategy.amounts, networkTokenConversionAmounts, totalGovTokenAmountToBurn);
+    }
+
+    /**
+     * @dev returns the result of the conversion the provided tokens to the governance token and its burn
+     *
+     * @param tokens the tokens to convert
+     *
+     * @return the amounts of the converted tokens
+     * @return the network token amounts the tokens were converted to
+     * @return the total burned amount in this vortex
+     * @return the incentive fee resulting from this vortex
+     */
+    function availableVortex(IERC20[] calldata tokens)
```

Comment:
I don't quite understand the purpose of this function.
it's easy to see which tokens are available on etherscan as well as a good estimation of the the total $ value.
does it serve a specific need of the app? like showing how many gov tokens are available for burning?
not even sure we need to ever include the burning function in the app, it seems to me like a function shows all the arb opportunities... these actions are performed on chain and there's an incentive for it.

Reply:
This function was also explicitly requested by the product team as well, so if it's not needed - I'd love to remove it...

Reply:
ok, let's discuss this one offline with product

Reply:
so we decided to remove this one